pkg/state/chart_dependency_test.go

```diff
@@ -78,24 +79,47 @@ func TestContains(t *testing.T) {
 				ChartName:         "abc",
 				Repository:        "oci://localhost:5000/aaa",
 				VersionConstraint: "0.1.0",
+				Alias:             "abc_0-1-0",
 			},
 			deps: map[string][]unresolvedChartDependency{
 				"abc": {
 					{
 						ChartName:         "abc",
 						Repository:        "oci://localhost:5000/aaa",
 						VersionConstraint: "0.1.0",
+						Alias:             "abc_0-1-0",
 					},
 				},
 			},
 			expected: true,
 		},
+		{
+			name: "different chart version",
+			dep: unresolvedChartDependency{
+				ChartName:         "abc",
+				Repository:        "oci://localhost:5000/aaa",
+				VersionConstraint: "0.1.0",
```

Comment:
Shouldn't this be tied to the release using the same chart?
For example, imagine `elasticsearchA` and `elasticsearchB` depending on `elasticsearch` chart, shall we use `elasticsearchA` and `elasticsearchB`, so that you should be able to selectively upgrade the chart version on either A or B release?

Reply:
sorry. I didn't get you. could you post a case.

Reply:
@mumoshu

Reply:
Ah, sorry, I meant to leave a comment on L102.
In another way, I think we should have "release name" in the `Alias` field.

Reply:
do you think what we should do when version field is empty? @mumoshu

Reply:
Let me check..
We shall add cases to tests in this file, with ReleaseSpec.Version and/or unresolvedChartDependency.VersionConstraint being empty 🤔